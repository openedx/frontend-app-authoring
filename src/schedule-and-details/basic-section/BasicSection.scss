.basic-section {
  .basic-info-list {
    display: flex;
    justify-content: space-between;
    margin-bottom: 1.75rem;
    padding-inline-start: 0;
    list-style-type: none;
  }

  .pgn__pageBanner-component {
    @include pgn-box-shadow(1, "down");

    background-color: $white;
    border-radius: .375rem;
  }

  .pgn__pageBanner-content {
    align-items: flex-start;
    padding: 1.5rem;
    flex-direction: column;
  }

  .pgn__pageBanner-dismissButtonContainer {
    margin: 1.5rem 1.5rem 0 0;
    align-self: baseline;
  }

  .pgn__card-header-content {
    margin-top: .75rem;
  }
}
