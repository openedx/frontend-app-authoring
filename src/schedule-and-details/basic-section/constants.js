/* eslint-disable-next-line import/prefer-default-export */
export const INVITE_STUDENTS_LINK_ID = 'invite-students-link';
