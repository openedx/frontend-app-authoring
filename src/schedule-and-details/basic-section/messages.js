import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  basicTitle: {
    id: 'course-authoring.schedule.basic.title',
    defaultMessage: 'Basic information',
  },
  basicDescription: {
    id: 'course-authoring.schedule.basic.description',
    defaultMessage: 'The nuts and bolts of this course',
  },
  basicEmailIcon: {
    id: 'course-authoring.schedule.basic.email-icon',
    defaultMessage: 'Invite your students email icon',
  },
  courseOrganization: {
    id: 'course-authoring.schedule.basic.organization',
    defaultMessage: 'Organization',
  },
  courseNumber: {
    id: 'course-authoring.schedule.basic.course-number',
    defaultMessage: 'Course number',
  },
  courseRun: {
    id: 'course-authoring.schedule.basic.course-run',
    defaultMessage: 'Course run',
  },
  basicBannerTitle: {
    id: 'course-authoring.schedule.basic.banner.title',
    defaultMessage: 'Promoting your course with edX',
  },
  basicBannerText: {
    id: 'course-authoring.schedule.basic.banner.text',
    defaultMessage: 'Your course summary page will not be viewable until your course has been announced. To provide content for the page and preview it, follow the instructions provided by your Program Manager. Please note that changes here may take up to a business day to appear on your course summary page.',
  },
  basicPromotionButton: {
    id: 'course-authoring.schedule.basic.promotion.button',
    defaultMessage: 'Invite your students',
  },
});

export default messages;
