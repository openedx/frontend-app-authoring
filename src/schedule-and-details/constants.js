export const defaultEntranceExamMinimumScorePct = '50';
