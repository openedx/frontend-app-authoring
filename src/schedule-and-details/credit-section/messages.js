import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  creditTitle: {
    id: 'course-authoring.schedule.credit.title',
    defaultMessage: 'Course credit requirements',
  },
  creditDescription: {
    id: 'course-authoring.schedule.credit.description',
    defaultMessage: 'Steps required to earn course credit',
  },
  creditHelp: {
    id: 'course-authoring.schedule.credit.help',
    defaultMessage:
      'A requirement appears in this list when you publish the unit that contains the requirement.',
  },
  creditMinimumGrade: {
    id: 'course-authoring.schedule.credit.minimum-grade',
    defaultMessage: 'Minimum grade',
  },
  creditProctoredExam: {
    id: 'course-authoring.schedule.credit.proctored-exam',
    defaultMessage: 'Successful proctored exam',
  },
  creditVerification: {
    id: 'course-authoring.schedule.credit.verification',
    defaultMessage: 'ID Verification',
  },
  creditNotFound: {
    id: 'course-authoring.schedule.credit.not-found',
    defaultMessage: 'No credit requirements found.',
  },
});

export default messages;
