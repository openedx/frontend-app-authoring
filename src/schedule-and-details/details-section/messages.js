import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  detailsTitle: {
    id: 'course-authoring.schedule-section.details.title',
    defaultMessage: 'Course details',
  },
  detailsDescription: {
    id: 'course-authoring.schedule-section.details.description',
    defaultMessage: 'Provide useful information about your course',
  },
  dropdownLabel: {
    id: 'course-authoring.schedule-section.details.dropdown.label',
    defaultMessage: 'Course language',
  },
  dropdownHelpText: {
    id: 'course-authoring.schedule-section.details.dropdown.help-text',
    defaultMessage: 'Identify the course language here. This is used to assist users find courses that are taught in a specific language. It is also used to localize the \'From:\' field in bulk emails.',
  },
  dropdownEmpty: {
    id: 'course-authoring.schedule-section.details.dropdown.empty',
    defaultMessage: 'Select language',
  },
});

export default messages;
