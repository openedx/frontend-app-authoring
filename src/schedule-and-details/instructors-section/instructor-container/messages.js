import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  instructorNameLabel: {
    id: 'course-authoring.schedule-section.instructor.name.label',
    defaultMessage: 'Name',
  },
  instructorNameHelpText: {
    id: 'course-authoring.schedule-section.instructor.name.help-text',
    defaultMessage: 'Please add the instructor\'s name',
  },
  instructorNameInputPlaceholder: {
    id: 'course-authoring.schedule-section.instructor.name.input.placeholder',
    defaultMessage: 'Instructor name',
  },
  instructorTitleLabel: {
    id: 'course-authoring.schedule-section.instructor.title.label',
    defaultMessage: 'Title',
  },
  instructorTitleHelpText: {
    id: 'course-authoring.schedule-section.instructor.title.help-text',
    defaultMessage: 'Please add the instructor\'s title',
  },
  instructorTitleInputPlaceholder: {
    id: 'course-authoring.schedule-section.instructor.title.input.placeholder',
    defaultMessage: 'Instructor title',
  },
  instructorOrganizationLabel: {
    id: 'course-authoring.schedule-section.instructor.organization.label',
    defaultMessage: 'Organization',
  },
  instructorOrganizationHelpText: {
    id: 'course-authoring.schedule-section.instructor.organization.help-text',
    defaultMessage: 'Please add the institute where the instructor is associated',
  },
  instructorOrganizationInputPlaceholder: {
    id: 'course-authoring.schedule-section.instructor.organization.input.placeholder',
    defaultMessage: 'Instructor organization',
  },
  instructorBioLabel: {
    id: 'course-authoring.schedule-section.instructor.bio.label',
    defaultMessage: 'Biography',
  },
  instructorBioHelpText: {
    id: 'course-authoring.schedule-section.instructor.bio.help-text',
    defaultMessage: 'Please add the instructor\'s biography',
  },
  instructorBioInputPlaceholder: {
    id: 'course-authoring.schedule-section.instructor.bio.input.placeholder',
    defaultMessage: 'Instructor biography',
  },
  instructorPhotoLabel: {
    id: 'course-authoring.schedule-section.instructor.photo.label',
    defaultMessage: 'Photo',
  },
  instructorPhotoHelpText: {
    id: 'course-authoring.schedule-section.instructor.photo.help-text',
    defaultMessage: 'Please add a photo of the instructor (Note: only JPEG or PNG format supported)',
  },
  instructorPhotoInputPlaceholder: {
    id: 'course-authoring.schedule-section.instructor.photo.input.placeholder',
    defaultMessage: 'Instructor photo URL',
  },
  instructorDelete: {
    id: 'course-authoring.schedule-section.instructor.delete',
    defaultMessage: 'Delete',
  },
});

export default messages;
