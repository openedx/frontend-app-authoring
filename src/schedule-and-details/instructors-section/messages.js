import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  instructorsTitle: {
    id: 'course-authoring.schedule-section.instructors.title',
    defaultMessage: 'Instructors',
  },
  instructorsDescription: {
    id: 'course-authoring.schedule-section.instructors.description',
    defaultMessage: 'Add details about the instructors for this course',
  },
  instructorAdd: {
    id: 'course-authoring.schedule-section.instructors.add-instructor',
    defaultMessage: 'Add Instructor',
  },
});

export default messages;
