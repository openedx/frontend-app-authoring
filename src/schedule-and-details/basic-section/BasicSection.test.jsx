import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import { INVITE_STUDENTS_LINK_ID } from './constants';
import messages from './messages';
import BasicSection from '.';

describe('<BasicSection />', () => {
  const RootWrapper = (props) => (
    <IntlProvider locale="en">
      <BasicSection {...props} />
    </IntlProvider>
  );

  const props = {
    intl: {},
    org: 'foo org',
    courseNumber: 'bar number',
    run: 'foo run',
    lmsLinkForAboutPage: 'link://to',
    marketingEnabled: true,
    courseDisplayName: 'foo course',
    platformName: 'Your platform name here',
  };

  it('renders basic section successfully', () => {
    const { getByText } = render(<RootWrapper {...props} />);
    expect(getByText(messages.basicTitle.defaultMessage)).toBeInTheDocument();
    expect(
      getByText(messages.basicDescription.defaultMessage),
    ).toBeInTheDocument();
    expect(
      getByText(messages.courseOrganization.defaultMessage),
    ).toBeInTheDocument();
    expect(getByText(props.org)).toBeInTheDocument();
    expect(getByText(messages.courseNumber.defaultMessage)).toBeInTheDocument();
    expect(getByText(props.courseNumber)).toBeInTheDocument();
    expect(getByText(messages.courseRun.defaultMessage)).toBeInTheDocument();
    expect(getByText(props.run)).toBeInTheDocument();
  });

  it('shows the page banner if the marketingEnabled is true', () => {
    const { getByText, queryAllByText } = render(<RootWrapper {...props} />);
    expect(
      getByText(messages.basicBannerTitle.defaultMessage),
    ).toBeInTheDocument();
    expect(
      getByText(messages.basicBannerText.defaultMessage),
    ).toBeInTheDocument();
    expect(queryAllByText('Course summary page').length).toBe(0);
  });

  it('shows the course promotion if the marketingEnabled is false', () => {
    const initialProps = { ...props, marketingEnabled: false };
    const { getByText, getByRole, queryAllByText } = render(
      <RootWrapper {...initialProps} />,
    );
    const inviteButton = getByRole('button', {
      name: messages.basicPromotionButton.defaultMessage,
    });

    expect(getByText(/Course Summary Page/i)).toBeInTheDocument();
    expect(
      getByText(/(for student enrollment and access)/i),
    ).toBeInTheDocument();
    expect(getByText(props.lmsLinkForAboutPage)).toBeInTheDocument();
    expect(inviteButton).toBeInTheDocument();
    expect(queryAllByText(messages.basicBannerTitle.defaultMessage).length).toBe(0);
  });

  it('checks link link to invite', () => {
    const initialProps = { ...props, marketingEnabled: false };
    const { getByTestId } = render(<RootWrapper {...initialProps} />);
    const inviteLink = getByTestId(INVITE_STUDENTS_LINK_ID);
    expect(decodeURIComponent(inviteLink.href)).toEqual(
      `mailto:${process.env.INVITE_STUDENTS_EMAIL_TO}?body=The course ${props.courseDisplayName}, provided by ${props.platformName}, is open for enrollment. Please navigate to this course at ${props.lmsLinkForAboutPage} to enroll.&subject=Enroll in ${props.courseDisplayName}.`,
    );
  });
});
