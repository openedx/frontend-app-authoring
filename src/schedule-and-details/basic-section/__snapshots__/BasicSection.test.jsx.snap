// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicSection /> matches the snapshots 1`] = `
<section
  className="section-container basic-section"
>
  <header
    className="section-header"
  >
    <span
      className="lead"
    >
      Basic information
    </span>
    <span
      className="x-small text-gray-700"
    >
      The nuts and bolts of this course
    </span>
  </header>
  <ul
    className="basic-info-list"
  >
    <li
      className="d-grid"
    >
      <h4
        className="text-gray-700"
      >
        Organization
      </h4>
      <span
        className="small"
      >
        foo org
      </span>
    </li>
    <li
      className="d-grid"
    >
      <h4
        className="text-gray-700"
      >
        Course number
      </h4>
      <span
        className="small"
      >
        bar number
      </span>
    </li>
    <li
      className="d-grid"
    >
      <h4
        className="text-gray-700"
      >
        Course run
      </h4>
      <span
        className="small"
      >
        foo run
      </span>
    </li>
  </ul>
  <div
    aria-atomic="true"
    aria-live="polite"
    className="pgn__pageBanner-component pgn__pageBanner__accentA"
    role="alert"
  >
    <div
      className="pgn__pageBanner-content"
    >
      <h4
        className="mb-2"
      >
        Promoting your course with edX
      </h4>
      <span
        className="text text-gray-700 text-left"
      >
        Your course summary page will not be viewable until your course has been announced. To provide content for the page and preview it, follow the instructions provided by your Program Manager. Please note that changes here may take up to a business day to appear on your course summary page.
      </span>
    </div>
    <span
      className="pgn__pageBanner-dismissButtonContainer"
    >
      <button
        aria-label="Dismiss"
        className="btn-icon btn-icon-black btn-icon-inline"
        onClick={[Function]}
        type="button"
      >
        <span
          className="btn-icon__icon-container"
        >
          <span
            className="pgn__icon btn-icon__icon"
          >
            <svg
              aria-hidden={true}
              fill="none"
              focusable={false}
              height={24}
              role="img"
              viewBox="0 0 24 24"
              width={24}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
                fill="currentColor"
              />
            </svg>
          </span>
        </span>
      </button>
    </span>
  </div>
</section>
`;
